.cirrus.yml

```diff
@@ -45,6 +45,18 @@ task:
         - if [[ "$CHANNEL" -eq "stable" ]]; then find . | grep _web$ | xargs rm -rf; fi
         - flutter channel $CHANNEL
         - ./script/build_all_plugins_app.sh apk
+    - name: e2e_web_smoke_test
+      # Tests e2e example test in web.
+      only_if: "changesInclude('.cirrus.yml', 'packages/e2e/**') || $CIRRUS_PR == ''"
+      script:
+        - flutter config --enable-web
+        - git clone https://github.com/flutter/web_installers.git
+        - cd web_installers/packages/web_drivers/
+        - pub get
+        - dart lib/web_driver_installer.dart &
+        - sleep 20
```

Comment:
Can we make this more deterministic? Maybe look at the output of web_driver_installer and use that to decide when to call flutter drive?

https://superuser.com/questions/375223/watch-the-output-of-a-command-until-a-particular-string-is-observed-and-then-e

Reply:
Hmmm... Why is the previous command run in the background, though?

Wouldn't you be able to remove the `sleep` if the `&` at the end of `dart lib/web_driver_installer.dart` were removed?

Reply:
No it should always run on the background. The code is installing (downloading over the web and extracting) the chrome driver. It also starts running it on port 4444

Reply:
Ahhh... maybe splitting the code in "the part that downloads and installs" (being synchronous) and the part that "starts the driver on port 444 in the background" could be split in two separate commands, that way you wouldn't need the sleep (maybe)

Reply:
What I can do is, I can only use web_driver installers for installing. and for launching I can use
chromedriver --port=4444 &

This can remove the need for sleep since it is  starts instantly.

Reply:
If that works, I think it's more deterministic than having the sleep there (imagine that installing takes 21 seconds for whatever reason)

Reply:
@ditman I send you a PR to fix the webdriver command (https://github.com/flutter/web_installers/pull/4)

Reply:
@nturgut just approved it!

Reply:
done.